examples/app/export_csv/download.js

```diff
@@ -1,26 +1,34 @@
-var data = source.data;
-var filetext = 'name,income,years_experience\n';
-for (var i = 0; i < data['name'].length; i++) {
-    var currRow = [data['name'][i].toString(),
-                   data['salary'][i].toString(),
-                   data['years_experience'][i].toString().concat('\n')];
+function table_to_csv(data_table) {
+    const columns = Object.keys(data_table)
+    const nrows = data_table[columns[0]].length
+    const lines = [columns.join(',')]
 
-    var joined = currRow.join();
-    filetext = filetext.concat(joined);
+    for (let i = 0; i < nrows; i++) {
+        let row = [];
+        for (let j = 0; j < columns.length; j++) {
+            const column = columns[j]
+            row.push(data_table[column][i].toString())
+        }
+        lines.push(row.join(','))
+    }
+    filetext = lines.join('\n').concat('\n')
+    return filetext
 }
 
-var filename = 'data_result.csv';
-var blob = new Blob([filetext], { type: 'text/csv;charset=utf-8;' });
+
+const data = source.data
+const filename = 'data_result.csv'
+filetext = table_to_csv(data)
```

Comment:
first and third lines can be combined:
```
const filetext = table_to_csv(source.data)
```

Reply:
Using `const` (or `let` for that matter) here gives me the javascript error: `Uncaught ReferenceError: filetext is not defined`